### PR TITLE
Rename Authenticator.white/blacklist to allowed/blocked

### PR DESCRIPTION
**Pull request by minrk**
cf https://github.com/jupyterhub/team-compass/issues/299

todo:

- [x] Authenticator.whitelist -> Authenticator.allowed_users
- [x] Authenticator.blacklist -> Authenticator.blocked_users
- [x] group_whitelist -> allowed_groups
- [x] Authenticator.check_whitelist -> Authenticator.check_allowed (this is the top-level method that checks *all* sources of allow—users, groups, etc.)
- [x] LocalAuthenticator.check_group_whitelist -> LocalAuthenticator.check_allowed_groups
- [x] deprecate traits, config
- [x] test deprecated names still work
- [x] test various combinations of deprecated config for proper priority resolution
- [x] subclass resolution in deprecated method names (subclass overriding check_whitelist should take priority over base class check_allowed, but warn about deprecation *unless* check_allowed is also defined on the subclass)
- [x] update docs

### CODE REVIEW

**Comment by minrk** [comment]
I believe this is ready to go now

File: jupyterhub/auth.py [hunk 1]
```diff
@@ -100,41 +101,74 @@ class Authenticator(LoggingConfigurable):
         """
     ).tag(config=True)
 
-    whitelist = Set(
+    whitelist = Set(help="Deprecated, use `Authenticator.allowed`", config=True,)
+
+    allowed = Set(
```
Comment (manics) [inline jupyterhub/auth.py:106]: Some authenticators have additional properties such as `allowed_groups`. Any thoughts on changing this to `allowed_users` for clarity?

**Comment by minrk** [comment]
What do people think abut the `.check_allowed` method name? This is the main method for checking if a user is allowed by any mechanism and is overridden by subclasses to also call `.check_allowed_groups()` etc. (see the LocalAuthenticator here that does just that), so I feel it's appropriate to be generic. I thought about `.check_allowed_users` to match the config name, but found that confusing that it should be overridden to check other sources of information, not just the `.allowed_users` set.

File: jupyterhub/auth.py (hunk 1 above)
Comment (minrk) [reply jupyterhub/auth.py]: Good idea. Switched to `allowed_users` pending resolution of [the discussion on team-compass](https://github.com/jupyterhub/team-compass/issues/299#issuecomment-648759244)

**Comment by minrk** [comment]
Ready for review again

File: jupyterhub/auth.py [hunk 2]
```diff
@@ -260,39 +294,74 @@ def my_hook(authenticator, handler, authentication):
...
         super().__init__(**kwargs)
-        for method_name in (
-            'check_whitelist',
-            'check_blacklist',
-            'check_group_whitelist',
+        self._init_deprecated_methods()
+
+    def _init_deprecated_methods(self):
+        # TODO: properly handle deprecated signature *and* name
+        # with correct subclass override priority!
```
Comment (GeorgianaElena) [inline jupyterhub/auth.py:301]:
```suggestion
        # handles deprecated signature *and* name
        # with correct subclass override priority!
```
not a todo anymore :tada:

Comment (minrk) [reply jupyterhub/auth.py]: Thanks!

**Pull request merged**